### PR TITLE
chore: Add filtered flag to EventEnvelope

### PR DESCRIPTION
**Pull request by patriknw**
* Needed for Projection over gRPC to be able see if filtered event comes from backtracking or not
* Previously we hacked around this by using eventMetadata= Some(NotUsed) but it didn't work out because we can't see the difference of backtracking or not

See https://github.com/akka/akka-persistence-r2dbc/pull/348

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala [hunk 1]
```diff
@@ -58,7 +82,21 @@ final class EventEnvelope[Event](
     val timestamp: Long,
     val eventMetadata: Option[Any],
     val entityType: String,
-    val slice: Int) {
+    val slice: Int,
+    val filtered: Boolean) {
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:86]: Doh, this is still not covering it. We still want to use `eventEnvelope=None` for the filtered events, so can still not see that it is from backtracking or not

Comment (johanandren) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:86]: Could we do an additional set/list of name of "flags" for producers and consumers to use if they want?

Comment (patriknw) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:86]: I added a `source: String` field, see usage in https://github.com/akka/akka-persistence-r2dbc/pull/348/commits/b73eba234d2a62701bd9c958e825d40f47106417

File: akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala [hunk 2]
```diff
@@ -58,7 +81,22 @@ final class EventEnvelope[Event](
...
     val eventMetadata: Option[Any],
     val entityType: String,
-    val slice: Int) {
+    val slice: Int,
+    val filtered: Boolean,
+  val source: String) {
+
+  // backwards compatibility when adding filtered and source
+  @deprecated("Use constructor with all parameters", "2.8.0-M4")
+  def this(
```
Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:90]: Do we need to deprecate this, I think for journals not doing filtering and backtracking this constructor is fine?

Comment (patriknw) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:90]: We have `apply` and `create` factory methods and there I kept the old signatures. I agree, no need to deprecate this constructor.

**Pull request merged**